### ISSUE TITLE
BTRoot doesn't work

### ISSUE THREAD

**Issue by Jedah85** [AdrienQuillet/godot-yet-another-behavior-tree#1]
I'm not sure if I'm doing something wrong, but I installed the addon into a new project, activated it into the project -> plugin settings. Then restarted Godot to be sure and when I go to the add node, it shows all the nodes of this addon but when choosing BTRoot it isn't added to the scene. I tried adding other nodes, and they all seem to work, but not BTRoot which is the most important one. I'm using the Godot 4.beta8 version

**Comment by Jedah85** [comment]
I now runned Godot using the Console and these relevant message showed up:

```
SCRIPT ERROR: Parse Error: Invalid operands "String" and "StringName" for "+" operator.
          at: GDScript::reload (res://addons/yet_another_behavior_tree/src/Nodes/BTRoot.gd:72)
```

And at the moment I try to add the BTRoot node into a scene:

```
ERROR: Cannot get class ''.
   at: (core/object/class_db.cpp:326)
ERROR: Parameter "p_object" is null.
   at: instantiate_object_properties (editor/editor_data.cpp:551)
ERROR: Condition "!child" is true.
   at: _do_create (editor/scene_tree_dock.cpp:2200
```

Hope this error messages help in finding the solution, as it looks like it's an amazing addon and I would like to use it. Thanks

**Comment by Earewien** [comment]
Thanks for reporting this issue :)!

I created this addon using Godot 4 beta 10. From Godot 4 beta 9/10 (don't known exactly), `StringName` can be concatenate to `String` using `+` operator.

Unfortunately, in Godot 4 beta 8, it is not possible. 

I changed the way to create the faulty string, so it should be safe now ! Available in release 1.0.1 You can download it here : https://github.com/AdrienQuillet/godot-yet-another-behavior-tree/releases/tag/1.0.1 and replace your original addon code by the new one, or patch the code yourself by reporting my [fix](https://github.com/AdrienQuillet/godot-yet-another-behavior-tree/commit/74e99948ca6e98772da13dc83f300ed36bb0d101) to your local addon-file. 

Hope this will help 😉
